theseus/labs/lie/functional/se3_impl.py

```diff
@@ -743,23 +757,24 @@ def backward(cls, ctx, grad_output):
 def _transform_from_impl(group: torch.Tensor, tensor: torch.Tensor) -> torch.Tensor:
     check_group_tensor(group)
     check_transform_tensor(tensor)
-    ret = group[:, :, -1:] + group[:, :, :3] @ tensor.view(-1, 3, 1)
-    return ret.reshape(tensor.shape)
+    ret = group[..., -1:] + group[..., :3] @ tensor.unsqueeze(-1)
+    return ret.squeeze(-1)
 
 
 def _jtransform_from_impl(
     group: torch.Tensor, tensor: torch.Tensor
 ) -> Tuple[List[torch.Tensor], torch.Tensor]:
     check_group_tensor(group)
     check_transform_tensor(tensor)
-    jacobian_g = group.new_empty(group.shape[0], 3, 6)
-    jacobian_g[:, :, :3] = group[:, :, :3]
-    jacobian_g[:, :, 3:] = -group[:, :, :3] @ SO3._hat_autograd_fn(tensor)
-    jacobian_p = group[:, :, :3].view(tensor.shape[:-1] + (3, 3))
-    jacobians = []
-    jacobians.append(jacobian_g)
-    jacobians.append(jacobian_p)
-    return jacobians, _transform_from_impl(group, tensor)
+    ret = _transform_from_impl(group, tensor)
+    size = get_transform_tensor_size(ret)
+    jacobian_g = group.new_empty(*size, 3, 6)
+    jacobian_g[..., :3] = group[..., :3]
+    jacobian_g[..., 3:] = -group[..., :3] @ SO3._hat_autograd_fn(tensor)
+    jacobian_p = group[..., :3]
+    jacobian_g = jacobian_g.expand(*size, 3, 6)
```

Comment:
I think the use of `expand` here could be problematic. We don't want shared memory for an output tensor that represents the outcome of a math operation. In fact, if you change `group[..., :3]` after you return from this function, the returned value of `jacobian_p[...,:3]` will also change (same for `jacobian_g`). We should use `torch.repeat`.

Reply:
Fixed.